ghcr.io.conf.template

```diff
@@ -1,36 +1,39 @@
-# PROXY_DOMAIN: user visible domain, e.g. docker.opensafely.org, which hosted by cloudflare
-# PROXY_ORIGIN: the backend, e.g. docker-proxy.dokku2.ebmdatalab.org
-# RESOLVER: DNS resolver for dynamically looking up redirect domains
-# PORT: provided by dokku, defaults to 80
+# Read-only proxy certain organisation to ghcr.io
+#
+# RESOLVER:a DNS resolver for dynamically looking up redirect domains. Defaults to 127.0.0.1
+# PORT: defaults to 8080, but dokku can override
 
 # The docker cli uses the realm domain given in the www-authenticate to auth.
 # So we need to re-write from ghcr.io to our proxy.
 # Use a map as it is built in, even though its not a clean way to replace strings.
 # Ideally would use more-headers module, but its not in the default image.
 map $upstream_http_www_authenticate $auth_header {
         # used to have 1 occurance of domain, now has 2. Handle both.
-        ~(.*)ghcr.io(.*)ghcr.io(.*) $1${PROXY_DOMAIN}$2${PROXY_DOMAIN}$3;
-        ~(.*)ghcr.io(.*)            $1${PROXY_DOMAIN}$2;
+        ~(.*)ghcr.io(.*)ghcr.io(.*) $1docker-proxy.opensafely.org$2docker-proxy.opensafely.org$3;
+        ~(.*)ghcr.io(.*)            $1docker-proxy.opensafely.org$2;
 }
 
+# these need to be global, sadly
 proxy_http_version 1.1; # keep alives to upstream
 proxy_buffering off;    # start streaming response immediately
-#error_log /var/log/nginx/error-debug.log debug;
 
 server {
 
-    # be identified by either domain
-    server_name ${PROXY_DOMAIN} ${PROXY_ORIGIN};
+    # note: embdatalab domain is for when we are being proxied by cloudflare
```

Comment:
Not sure I quite follow - do you mean "the ebmdatalab domain allows you direct access to the server when the primary domain is being proxied by CloudFlare"?

Reply:
Kinda - it's so nginx picks the correct vhost when cloudflare is proxying `docker-proxy.opensafely.org `to the origin domain of `docker-proxy.ebmdatalab.net` (as we are currently). So the vhost can respond to either domain. When EMIS add our new floating IP, plan is to disable proxying, and have nginx host the `docker-proxy.opensafely.org` domain directly. Hence why it has both.

Reply:
ah ok, let's tidy when EMIS add this ip then